src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -543,6 +580,37 @@ SGVector<T> rowwise_sum(const Block<SGMatrix<T>>& a, bool no_diag=false)
 	return sg_linalg->get_cpu_backend()->rowwise_sum(a, no_diag);
 }
 
+/**
+ * Generic linear solver
+ *
+ * @param A left-hand side matrix
+ * @param b right-hand side array
+ * @param solver_type solver method 1: Cholesky solver
+ * @return array x
+ */
+template <typename T>
+SGVector<T> solver(const SGMatrix<T>& A, const SGVector<T>& b,
+	LINALG_SOLVER_TYPE solver_type)
+{
+	SGMatrix<T> L;
+	SGVector<T> result;
+	switch (solver_type)
+	{
+		/* Solve the linear equations \f$Ax=b\f$,
+		 * where \f$A\f$ is a Hermitian positive definite matrix
+		*/
+		case ST_CHOLESKY:
```

Comment:
You think it is better to directly call eigen3 here? Then some overhead is avoided.... (storing the lower factor)